### PR TITLE
Refactor ModuleSearchBox into reusable SearchBox

### PR DESCRIPTION
**Pull request by taneliang**
I've refactored the UI and style of ModuleSearchBox into a new SearchBox so that the search box will feel and work consistently in both the Venues page ([WIP](https://github.com/taneliang/nusmods/tree/venue-info-page)) and the Browse page.

This is in a separate pull request for easier review :)

### CODE REVIEW

File: v3/src/js/views/browse/ModuleSearchBox.jsx [hunk 1]
```diff
@@ -3,113 +3,49 @@ import type { ContextRouter } from 'react-router-dom';
...
-        {!this.props.useInstantSearch && this.state.hasChanges && <p className={styles.searchHelp}>
-          Press enter to search
-        </p>}
-      </div>
+      <SearchBox
+        throttle={this.props.throttle}
+        useInstantSearch={this.props.useInstantSearch}
+        initialSearchTerm={this.props.initialSearchTerm}
+        placeholder="Module code, names and descriptions"
+        onSearch={this.search}
```
Comment (ZhangYiJiang) [inline v3/src/js/views/browse/ModuleSearchBox.jsx]: Can bind `this.props.searchModules` directly

File: v3/src/js/views/browse/ModuleSearchBox.jsx [hunk 2]
```diff
@@ -3,113 +3,49 @@ import type { ContextRouter } from 'react-router-dom';
...
-      this.setState({ searchTerm, hasChanges: true });
-
-      if (this.props.useInstantSearch) this.debouncedSearch(searchTerm);
-    }
-  };
-
   search = (input: string) => {
-    this.setState({ hasChanges: false });
-    this.props.searchModules(input.trim());
+    this.props.searchModules(input);
```
Comment (ZhangYiJiang) [inline v3/src/js/views/browse/ModuleSearchBox.jsx]: Can be bound to `onSearch` directly since

Comment (taneliang) [reply v3/src/js/views/browse/ModuleSearchBox.jsx]: Oops good point

File: v3/src/js/views/browse/ModuleSearchBox.jsx (hunk 1 above)
Comment (taneliang) [reply v3/src/js/views/browse/ModuleSearchBox.jsx]: Oops good point

**Pull request merged**